src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.FixAllProvider.cs

```diff
@@ -26,7 +26,7 @@ public async override Task<CodeAction> GetFixAsync(FixAllContext fixAllContext)
                     return null;
                 }
 
-                var batchFixer = WellKnownFixAllProviders.BatchFixer;
```

Comment:
❔  Why make this change?

Reply:
See https://github.com/dotnet/roslyn/pull/41363/files#r376485372

Reply:
> This change was needed as the GetFixAsync API below was moved from being an internal virtual API on public FixAllProvider type to an internal API directly on BatchFixAllProvider as this was the only place that API was being invoked.

Reply:
Why not just call the public API?

Reply:
Yeah, we can update this to use the public API and it literally does exactly the same thing, but with less code and less coupling.

Reply:
Likely some perf optimization. Also note that this type works for multi-select and suppress error list diagnostics feature, so likely needs custom set of diagnostics to be passed in. In any case, seems like that there might be further cleanup that can be done in this area, and creating a separate tracking item for that work seems reasonable.

Reply:
> Likely some perf optimization. 

I mean it does _exactly_ the same thing. The same calls, on the same objects, in the same order.

Reply:
> Yeah, we can update this to use the public API and it literally does exactly the same thing, but with less code and less coupling.

I am concerned it will break "multi-select and suppress error list diagnostics feature". It seems to do this in a separate PR which is not blocking #41363

Reply:
If the goal of this PR is fixing the calls to internal `FixAllProvider.GetFixAsync` overloads, I think we should go ahead and apply this fix at the same time.

Reply:
Goal of this PR is to get rid of the internal virtual members on the public types `FixAllProvider` and `FixAllProvider.DiagnosticProvider`, which is needed to move some code into shared layer.

Reply:
Here are the two paths for the first internal virtual method:

https://github.com/dotnet/roslyn/blob/0dc28dc46c494189d1aca48224fa51eca548e8bd/src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs#L35-L38

https://github.com/dotnet/roslyn/blob/0dc28dc46c494189d1aca48224fa51eca548e8bd/src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.FixAllProvider.cs#L43-L50

Reply:
Here are the two paths for the second internal virtual method:

https://github.com/dotnet/roslyn/blob/0dc28dc46c494189d1aca48224fa51eca548e8bd/src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs#L40-L43

https://github.com/dotnet/roslyn/blob/0dc28dc46c494189d1aca48224fa51eca548e8bd/src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.FixAllProvider.cs#L53-L60

Reply:
Do we know that `fixAllContext.State` passed into the internal virtual API is identical to what will be computed when invoking the public API?

Reply:
The specific suggestion here is:

```csharp
if (fixAllContext.Document != null)
{
  return !isGlobalSuppression
    ? await batchFixer.GetFixAsync(fixAllContext).ConfigureAwait(false)
    : GlobalSuppressMessageFixAllCodeAction.Create(
        title,
        suppressionFixer,
        fixAllContext.Document,
        await fixAllContext.GetDocumentDiagnosticsToFixAsync().ConfigureAwait(false));
}
else
{
  // similar...
}
```

Reply:
happy to try this!

Reply:
Yes, that sounds reasonable. @CyrusNajmabadi can you also validate that this does not break the error list feature where you multi-select diagnostics and then right click to select suppress will add suppressions for selected diagnostics? I don't believe we have tests for that feature.

Reply:
yes, will do.

Reply:
Yup.  I've validated that both "Suppress in source" and "suppress in suppression file" work.